### ISSUE TITLE
Whilelist keeps reseting to default

### ISSUE THREAD

**Issue by Xalcer** [rNeomy/auto-tab-discard#208]
Hi.

I just installed the extension on Chrome and works as expected, except the whitelist doesn't last long. After a few minutes, it goes blank again, ignoring whatever I've set.

**Comment by rNeomy** [comment]
So if you add a new URL to the list and press the save button and close the options page, on the next open, you don't see the URL?

**Comment by Xalcer** [comment]
If I open again inmediately it's still saved. It's after a few minutes or when I close and open the browser when it resets.

**Comment by rNeomy** [comment]
do you have any policy for this extension? chrome://policy/

**Comment by Xalcer** [comment]
No policies established.

**Comment by rNeomy** [comment]
I am not sure what could cause this. Can you try the extension on a clean profile?

**Comment by inkhsutesou** [comment]
![image](https://user-images.githubusercontent.com/4542497/114900644-124ae500-9de2-11eb-8a5b-5ad97ab51767.png)
Same issue

**Comment by rNeomy** [comment]
The options page loads settings once it gets opened. So if you have it opened while you add a new hostname from the popup interface, you will need to refresh the page to see updates. Also, there is no need to save anything if you add hostnames from the popup interface.

**Comment by Soraph** [comment]
I have the same issue even after the latest update, it discard everything and disregard the settings like pinned tabs and whitelist, if I reboot the browser I have the whitelist empty

**Comment by rNeomy** [comment]
Let's debug:

1. Open "chrome://extensions/" in a browser tab and turn the developer mode on
2. Find the background page of this extension and switch to the Console tab
3. Type "chrome.runtime.reload()" and see if you get an error on the background page.

**Comment by Soraph** [comment]
No errors

**Comment by rNeomy** [comment]
@Soraph can you reproduce this on the latest release? v0.4.6

**Comment by Soraph** [comment]
@rNeomy I can confirm that the issue is resolved with 0.4.6, I just checked with a browser restart and an OS restart (Brave on Win10). The options are persisted correctly and the tabs seems to behave as expected

**Comment by rNeomy** [comment]
perfect! @Xalcer can you confirm too?

**Comment by Xalcer** [comment]
It seems to work correctly now.

**Comment by endymion0000** [comment]
There's still problems with the Exceptions whitelist in 0.4.6.

It still occasionally wipes the list and just now I had 2 domains in it from a previous session. I went to add about 4 or 5 new ones by visiting the sites one after the other and selecting Keep Tabs For This Site: Always. Reloaded the plugin options and the only one listed was the last one I did. The 2 were gone along with all of  the new ones except that last one.

So while the list can persist, it may be the adding to the list process that sometimes is going awry.
